misk-redis/src/main/kotlin/misk/redis/RealRedis.kt

```diff
@@ -60,6 +60,13 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
     }
   }
 
+  // Set a ByteArray value if it doesn't already exist
+  override fun setnx(key: String, value: ByteString) {
```

Comment:
could you add this method accepting a TTL value
```
override fun setnx(key: String, expiryDuration: Duration, value: ByteString)
```

Reply:
so the setnx function doesn't have an expiration field

Reply:
at least the one in jedis doesn't